parsec/api/protocol/types.py

```diff
@@ -98,6 +98,22 @@ def new(cls: Type[DeviceIDTypeVar]) -> DeviceIDTypeVar:
         return cls(f"{uuid4().hex}@{uuid4().hex}")
 
 
+class DeviceLabel(str):
+    def __new__(cls, device_label: str) -> "DeviceLabel":
+        device_label = normalize("NFC", device_label)
+
+        if not 0 < _bytes_size(device_label) < 255:
+            raise ValueError("Invalid device_label")
+
+        return super(DeviceLabel, cls).__new__(cls, device_label)
+
+
+class DeviceLabelField(fields.String):  # type: ignore
```

Comment:
`# type: ignore` for avoid mypy strange error (only quick fix): 
```
parsec/api/protocol/types.py:111: error: Class cannot subclass 'String' (has type 'Any')  [misc]
parsec/api/protocol/types.py:111: error: Base type String becomes "Any" due to an unfollowed import  [no-any-unimported]
```

Reply:
why not using `str_based_field_factory` ?

```python
DeviceLabelField = fields.str_based_field_factory(DeviceLabel)
```